### ISSUE TITLE
OData library 8.0.4 does not add the control information property for open types. (using odata.metadata=minimal accept header value)

### ISSUE THREAD

**Issue by BenAlfasi** [OData/AspNetCoreOData#457]
Hi,

Note - we get the following Accept header: accept=application/json;odata.metadata=minimal)
When we get the full accept header (accept=application/json;odata.metadata=full), we have control information for all properties (except strings of course).

Example:

Created simple Open type:
```C#
public class HuntingRowResults
{
public IDictionary<string, object> DynamicProperties { get; set; }
}
```
when I'm adding int64 to the dictionary I'm getting the following response:
```json
{
"EmailClusterId": 2629759514
}
```
But the expected response should be:
```json
{
"EmailClusterId@odata.type": "#Int64",
"EmailClusterId": 2629759514
}
```
Thanks,
Ben

**Comment by xuzhg** [comment]
To unblock you, I create a sample for your reference at: https://github.com/xuzhg/WebApiSample/tree/main/v8.x/AddTypeAnnotationExtensions

You can download it, run, and send the following request:
```c#
http://localhost:5134/odata/customers
```
you can get:
```json
{
    "@odata.context": http://localhost:5134/odata/$metadata#Customers,
    "value": [
        {
            "Id": 1,
            "Name": "Bracing",
            "DynamicLong@odata.type": "#Int64",
            "DynamicLong": 44,
            "DynamicBool": false,
            "DynamicString": "abc"
        },
        {
            "Id": 2,
            "Name": "Chilly",
            "DynamicGuid@odata.type": "#Guid",
            "DynamicGuid": "6c0aad01-b8a7-45f7-9e8c-c5dad40b8d5b",
            "DynamicByte@odata.type": "#Byte",
            "DynamicByte": 5
        }
    ]
}
```

**Comment by habbes** [comment]
@xuzhg is there documentation on extending the serializer?

**Comment by habbes** [comment]
This seems related to https://github.com/OData/AspNetCoreOData/issues/451

**Comment by xuzhg** [comment]
@BenAlfasi Can you try this nightly: Microsoft.AspNetCore.OData.8.0.6-Nightly202201312349

For how to setup visual studio and config the nightly, you can refer to: https://github.com/OData/AspNetCoreOData#34-nightly-builds

**Comment by BenAlfasi** [comment]
@xuzhg  its working!
but note that now it is adding the minimal control information (Int 64, DateTimeOffset), by default, regardless if we add the minimal Odata metadata header (accept=application/json;odata.metadata=minimal)

**Comment by julealgon** [comment]
> @xuzhg its working! but note that now it is adding the minimal control information (Int 64, DateTimeOffset), by default, regardless if we add the minimal Odata metadata header (accept=application/json;odata.metadata=minimal)

@BenAlfasi I believe `application/json;odata.metadata=minimal` is actually the default for OData as per the spec. If you want no metadata, you can pass `application/json;odata.metadata=none` explicitly (or if you want complete metadata, `application/json;odata.metadata=full`).

**Comment by xuzhg** [comment]
@BenAlfasi  Shipped as 8.0.7. Please try and let me know any problems. Thanks.
